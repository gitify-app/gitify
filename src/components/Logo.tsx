interface IProps {
  isDark?: boolean;
  onClick?: () => void;
  className?: string;
}

const LIGHT_GRADIENT_START = '#CCCCCC';
const LIGHT_GRADIENT_END = '#FFFFFF';

const DARK_GRADIENT_START = '#22283B';
const DARK_GRADIENT_END = '#555B6E';

export const Logo = ({ isDark, onClick, className = '', ...props }: IProps) => (
  <svg
    className={className}
    onClick={() => onClick?.()}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    viewBox="0 0 500 500"
    role="img"
    aria-label="Gitify Logo"
    {...props}
  >
    <defs>
      <linearGradient
        x1="90.8779989%"
        y1="80.0676288%"
        x2="-1.30249747%"
        y2="32.7108237%"
        id={isDark ? 'linearGradient-dark' : 'linearGradient-light'}
      >
        <stop
          stopColor={isDark ? DARK_GRADIENT_START : LIGHT_GRADIENT_START}
          offset="0%"
        />
        <stop
          stopColor={isDark ? DARK_GRADIENT_END : LIGHT_GRADIENT_END}
          offset="100%"
        />
      </linearGradient>
    </defs>
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g
        id="Logo"
        transform="translate(6.000000, 6.000000)"
        fill={`url(${
          isDark ? '#linearGradient-dark' : '#linearGradient-light'
        })`}
        fillRule="nonzero"
      >
        <path d="M330.391053,10.2526316 C361.413773,10.2526316 386.562656,35.4015165 386.562656,66.4242368 C386.562656,97.4469571 361.413773,122.595841 330.391053,122.595842 L330.365421,122.570211 L247.319105,122.570211 C180.042515,122.499128 125.399186,176.890222 125.159,244.166421 L125.133789,244.219632 L125.154247,245.84774 C126.219609,301.255958 164.473457,349.116728 218.453292,362.268048 C272.940509,375.542983 329.466993,350.045988 355.583747,300.428701 L272.258632,300.427737 C257.346614,300.427737 243.046846,294.496977 232.512078,283.942976 C221.97731,273.388975 216.072618,259.078414 216.099748,244.166421 C216.099748,213.150778 241.242989,188.007632 272.258632,188.007632 L371.119632,188.007632 C379.28704,188.480415 386.935401,192.166372 392.393842,198.260263 L474.869789,300.324632 L474.978789,300.325211 L474.952789,300.426632 L474.953158,300.427737 L474.951789,300.427632 L474.173714,303.448935 C446.939817,406.220475 353.906452,478.028529 247.319105,478.080211 C118.007789,478.080211 12.8157895,373.144526 12.8157895,244.166421 C12.8157895,115.188316 117.905263,10.2526316 247.319105,10.2526316 L330.391053,10.2526316 Z" />
      </g>
    </g>
  </svg>
);
