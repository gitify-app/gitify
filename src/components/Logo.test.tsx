import { fireEvent, render, screen } from '@testing-library/react';

import * as TestRenderer from 'react-test-renderer';

import { Logo } from './Logo';

describe('components/Logo.tsx', () => {
  it('renders correctly (light)', () => {
    const tree = TestRenderer.create(<Logo />);

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly(dark)', () => {
    const tree = TestRenderer.create(<Logo isDark />);

    expect(tree).toMatchSnapshot();
  });

  it('should click on the logo', () => {
    const onClick = jest.fn();
    render(<Logo onClick={onClick} />);

    fireEvent.click(screen.getByLabelText('Gitify Logo'));

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
