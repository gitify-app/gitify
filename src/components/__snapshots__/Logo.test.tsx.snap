// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Logo.tsx renders correctly (light) 1`] = `
<svg
  aria-label="Gitify Logo"
  className=""
  onClick={[Function]}
  role="img"
  viewBox="0 0 500 500"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <linearGradient
      id="linearGradient-light"
      x1="90.8779989%"
      x2="-1.30249747%"
      y1="80.0676288%"
      y2="32.7108237%"
    >
      <stop
        offset="0%"
        stopColor="#CCCCCC"
      />
      <stop
        offset="100%"
        stopColor="#FFFFFF"
      />
    </linearGradient>
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      fill="url(#linearGradient-light)"
      fillRule="nonzero"
      id="Logo"
      transform="translate(6.000000, 6.000000)"
    >
      <path
        d="M330.391053,10.2526316 C361.413773,10.2526316 386.562656,35.4015165 386.562656,66.4242368 C386.562656,97.4469571 361.413773,122.595841 330.391053,122.595842 L330.365421,122.570211 L247.319105,122.570211 C180.042515,122.499128 125.399186,176.890222 125.159,244.166421 L125.133789,244.219632 L125.154247,245.84774 C126.219609,301.255958 164.473457,349.116728 218.453292,362.268048 C272.940509,375.542983 329.466993,350.045988 355.583747,300.428701 L272.258632,300.427737 C257.346614,300.427737 243.046846,294.496977 232.512078,283.942976 C221.97731,273.388975 216.072618,259.078414 216.099748,244.166421 C216.099748,213.150778 241.242989,188.007632 272.258632,188.007632 L371.119632,188.007632 C379.28704,188.480415 386.935401,192.166372 392.393842,198.260263 L474.869789,300.324632 L474.978789,300.325211 L474.952789,300.426632 L474.953158,300.427737 L474.951789,300.427632 L474.173714,303.448935 C446.939817,406.220475 353.906452,478.028529 247.319105,478.080211 C118.007789,478.080211 12.8157895,373.144526 12.8157895,244.166421 C12.8157895,115.188316 117.905263,10.2526316 247.319105,10.2526316 L330.391053,10.2526316 Z"
      />
    </g>
  </g>
</svg>
`;

exports[`components/Logo.tsx renders correctly(dark) 1`] = `
<svg
  aria-label="Gitify Logo"
  className=""
  onClick={[Function]}
  role="img"
  viewBox="0 0 500 500"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <linearGradient
      id="linearGradient-dark"
      x1="90.8779989%"
      x2="-1.30249747%"
      y1="80.0676288%"
      y2="32.7108237%"
    >
      <stop
        offset="0%"
        stopColor="#22283B"
      />
      <stop
        offset="100%"
        stopColor="#555B6E"
      />
    </linearGradient>
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      fill="url(#linearGradient-dark)"
      fillRule="nonzero"
      id="Logo"
      transform="translate(6.000000, 6.000000)"
    >
      <path
        d="M330.391053,10.2526316 C361.413773,10.2526316 386.562656,35.4015165 386.562656,66.4242368 C386.562656,97.4469571 361.413773,122.595841 330.391053,122.595842 L330.365421,122.570211 L247.319105,122.570211 C180.042515,122.499128 125.399186,176.890222 125.159,244.166421 L125.133789,244.219632 L125.154247,245.84774 C126.219609,301.255958 164.473457,349.116728 218.453292,362.268048 C272.940509,375.542983 329.466993,350.045988 355.583747,300.428701 L272.258632,300.427737 C257.346614,300.427737 243.046846,294.496977 232.512078,283.942976 C221.97731,273.388975 216.072618,259.078414 216.099748,244.166421 C216.099748,213.150778 241.242989,188.007632 272.258632,188.007632 L371.119632,188.007632 C379.28704,188.480415 386.935401,192.166372 392.393842,198.260263 L474.869789,300.324632 L474.978789,300.325211 L474.952789,300.426632 L474.953158,300.427737 L474.951789,300.427632 L474.173714,303.448935 C446.939817,406.220475 353.906452,478.028529 247.319105,478.080211 C118.007789,478.080211 12.8157895,373.144526 12.8157895,244.166421 C12.8157895,115.188316 117.905263,10.2526316 247.319105,10.2526316 L330.391053,10.2526316 Z"
      />
    </g>
  </g>
</svg>
`;
